File: Dante1/Module1.06/Zadanie6.13/main.c
#include <stdio.h>
#include <limits.h>

int main()
{
    int error;

    int array[5][5];

    printf("Enter:\n");
    for(int i = 0; i < 5; i++)
    {
        for(int j = 0; j < 5; j++)
        {
            error = scanf("%i", &array[i][j]);
            if(error != 1)
            {
                printf("Incorrect input");
                return 1;
            }
        }
    }

    int max = INT_MIN;

    for (int i = 0; i < 5; i++)
    {
        for (int j = 0; j < 5; j++)
        {
            for (int k = 0; k < 5; k++)
            {
                for (int l = 0; l < 5; l++)
                {
                    if (i != k || j != l)
                    {
                        int sum = array[i][j] + array[k][l];
                        if (sum > max)
                        {
                            max = sum;
                        }
                    }
                }
            }
        }
    }

    int counter = 0;
    for (int i = 0; i < 5; i++)
    {
        for (int j = 0; j < 5; j++)
        {
            for (int k = 0; k < 5; k++)
            {
                for (int l = 0; l < 5; l++)
                {
                    if ((i != k || j != l) && (i * 5 + j < k * 5 + l))
                    {
                        int sum = array[i][j] + array[k][l];
                        if (sum == max)
                        {
                            counter++;
                        }
                    }
                }
            }
        }
    }

    printf("%i %i\n", max, counter);
    for (int i = 0; i < 5; i++)
    {
        for (int j = 0; j < 5; j++)
        {
            for (int k = 0; k < 5; k++)
            {
                for (int l = 0; l < 5; l++)
                {
                    if ((i != k || j != l) && (i * 5 + j < k * 5 + l))
                    {
                        int sum = array[i][j] + array[k][l];
                        if (sum == max)
                        {
                            printf("%i %i %i %i\n", i, j, k, l);
                        }
                    }
                }
            }
        }
    }

    return 0;
}

File: Dante1/Module1.06/Zadanie6.13/main.rs
fn main() {
    let mut array: [[i32; 5]; 5] = [[0; 5]; 5];
    let mut input: String = String::new();

    println!("Enter: ");
    for i in 0..5 {
        for j in 0..5 {
            input.clear();
            std::io::stdin().read_line(&mut input).expect("Incorrect");
            array[i][j] = input.trim().parse().expect("Incorrect");
        }
    }

    let mut max: i32 = i32::MAX;
    for i in 0..5 {
        for j in 0..5 {
            for k in 0..5 {
                for l in 0..5 {
                    if i != k || j != l {
                        let sum: i32 = array[i][j] + array[k][l];
                        if sum > max {
                            max = sum;
                        }
                    }
                }
            }
        }
    }

    let mut counter: i32 = 0;
    for i in 0..5 {
        for j in 0..5 {
            for k in 0..5 {
                for l in 0..5 {
                    if (i != k || j != l) && (i * 5 + j < k * 5 + l) {
                        let sum: i32 = array[i][j] + array[k][l];
                        if sum == max {
                            counter += 1;
                        }
                    }
                }
            }
        }
    }

    println!("\n{} {}", max, counter);

    for i in 0..5 {
        for j in 0..5 {
            for k in 0..5 {
                for l in 0..5 {
                    if (i != k || j != l) && (i * 5 + j < k * 5 + l) {
                        let sum: i32 = array[i][j] + array[k][l];
                        if sum == max {
                            println!("{i} {j} {k} {l}");
                        }
                    }
                }
            }
        }
    }
}
